include/circt/Dialect/FIRRTL/FIRRTLOpInterfaces.h

```diff
@@ -64,22 +65,118 @@ struct PortInfo {
 /// Verification hook for verifying module like operations.
 LogicalResult verifyModuleLikeOpInterface(FModuleLike module);
 
-class InnerSymbolTable {
+namespace detail {
+LogicalResult verifyInnerRefs(Operation *op);
+} // namespace detail
```

Comment:
This looks normal, but any idea why this shouldn't be a private static function on `InnerRefNamespace`?

Reply:
Good question.  Looks like it's to avoid copies for every concrete type:

> Note: It is generally good practice to define the implementation of the verifyTrait or verifyRegionTrait hook out-of-line as a free function when possible to avoid instantiating the implementation for every concrete operation type.

[Source](https://mlir.llvm.org/docs/Traits/)